feat(argo-cd): Argo CD 1.7.11

hairmare opened the pull request:
* Update argo-cd to 1.7.11
* Move redis to new official repository instead of helm/stable: https://github.com/helm/charts/pull/21567
* Update redis-ha to the last 5.x version of redis (5.0.6)
* Update the bundled redis to 5.0.10 and use the redis alpine image as in https://github.com/argoproj/argo-cd/pull/4767

Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA
* [x] the build is green.
* [x] I will test my changes again once merged to master and published.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/argoproj/argo-helm?pullRequest=502) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/argoproj/argo-helm?pullRequest=502) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/argoproj/argo-helm?pullRequest=502) it.</sub>

hairmare commented:
chart-testing is complaining about "Error: requirements.lock is out of sync with requirements.yaml" but I can't reproduce an outdated lockfile locally.

I tried adding `https://dandydeveloper.github.io/charts/` to `.circleci/chart-testing.yaml` but reverted that change since it didn't help make tests green.

KrustyHack commented:
Ay @hairmare,

I think you forgot to update Chart.yaml "appVersion" to 1.7.9.

Anyway, I hope this MR will be merged soon ! Thanks for your work !

hairmare commented:
> I think you forgot to update Chart.yaml "appVersion" to 1.7.9.

Thanks for pointing that out @KrustyHack I fixed it but I don't think it'll make CI green.

I might try splitting this up into multiple PRs since i don't think the redis update needs to be in the same change. I do need to test if it argo still works if i don't include the redis update though.

KrustyHack commented:
Ay,

I've tested CircleCI build inside `gcr.io/kubernetes-charts-ci/test-image:v3.1.0` container and I did a `helm dep update` and it works.

Maybe a missing step on CircleCI definition ? I don't know well CircleCI.

nicolai92 commented:
@KrustyHack 

I had a quick look at the CircleCI pipeline and right after the Helm repository for ArgoCD is added, it states that the `requirements.yaml` file is out of sync. Either the `helm dep update` has to run manually before the commit or it must be added as a step to the CircleCI pipeline. IMO I would run the `helm dep update` on the local machine to have reproducible builds in the end.

spencergilbert commented:
Sorry catching up here, it sounds like @hairmare needs to run a `helm dep update` and commit those changes?

hairmare commented:
I just updated and rebased this PR.

> Sorry catching up here, it sounds like @hairmare needs to run a helm dep update and commit those changes?

I re-ran `helm dep update` but CI is still failing. 

I updated to Helm 3.4.2 & I'm not seeing any changes to the  lockfile that are not in this PR :(

Junnplus commented:
@hairmare reference https://github.com/argoproj/argo-helm/pull/520/files#diff-730178cdb8be754b6147b3b7278d70c7da616e59c1ec87c900e75ce41e5c708dR3

hairmare commented:
I rebased this once more out of curiosity and it still fails :sob: 

If we can get #520 ready to merge then I'll close this since I'd also rather upgrade to 1.8 as well.

spencergilbert commented:
> I rebased this once more out of curiosity and it still fails 
> 
> If we can get #520 ready to merge then I'll close this since I'd also rather upgrade to 1.8 as well.

Definitely agree - and still I have no idea what's going on with circle ci :/ -- I promise to stay on top of the 1.8.1 PR and get that merged in.

onyxet commented:
@hairmare Hello. Thank you for your work here. I'm wondering why we're using some private repository for redis-ha in argocd repo? What are the advantages over public repo like `Bitnami` or similar?
Will that repo `dandydeveloper.github.io/` will be supported by community as a https://artifacthub.io/ or bitnami charts?

CC: @KrustyHack

ailox commented:
In the mean time, 1.7.10, 1.7.11, 1.8.0 and 1.8.1 have been released. Should this PR be updated with the latest stable release, or should 4 new PRs be created?

hairmare commented:
> @hairmare Hello. Thank you for your work here. I'm wondering why we're using some private repository for redis-ha in argocd repo? What are the advantages over public repo like `Bitnami` or similar?
> Will that repo `dandydeveloper.github.io/` will be supported by community as a https://artifacthub.io/ or bitnami charts?

It's what was linked as the new official repo in the [deprecation warning on `stable/redis-ha`](https://github.com/helm/charts/tree/master/stable/redis-ha#deprecation-warning).

diranged commented:
Hi .. is this PR going to get finished up - and can someone work on the 1.8.2 release?

hairmare commented:
> Hi .. is this PR going to get finished up - and can someone work on the 1.8.2 release?

I'll rebase after #539 is merged. Once CI is healthy again, working on 1.8.2 should be easier.

hairmare commented:
Rebased on top of #539.

Looking back, the redis-ha chart update and the argo-cd should probably each be their own PR.

I plan on rolling out the 1.7.11 bump before switching to `1.8.*` so I'll work on a PR for `1.8.*` once this has landed.
